### ISSUE TITLE
When searching using a filter with "OR group", only results that satisfy all (not just any) conditions in OR group are returned

### ISSUE THREAD

**Issue by PolinaSannikova** [cuba-platform/cuba#1436]
### Environment

- Platform version: 6.8.13
- Database: PostgreSQL

### Description of the bug or enhancement

- Minimal reproducible example
Create filter with two "Assigned Driver" and "Assigned Vehicle" together in OR group, tick both, perform search
- Expected behavior
Device info is shown for devices with "Assigned Driver" or "Assigned Vehicle" (or both) not null 
- Actual behavior
Device info is shown for only for devices with both "Assigned Driver" "Assigned Vehicle" not null

We are using the following filter:

![default](https://user-images.githubusercontent.com/24522994/47420008-80bd0800-d78e-11e8-9626-fd492d73e04d.png)

![default](https://user-images.githubusercontent.com/24522994/47422071-e7dcbb80-d792-11e8-922d-896ee4e874f7.png)

When we put check marks in both "Assigned Driver" and "Assigned Vehicle", only device info for devices with both Assigned Driver and Assigned Vehicle are shown. The reason for that is the SQL query used when performing search with a filter - when check marks are put there, there are inner joins both for the TAXI_DRIVER table and TAXI_VEHICLE table, so only devices with both Assigned Driver and Assigned Vehicle are shown.
SQL queries attached - one with both  "Assigned Driver" and "Assigned Vehicle" ticked, the other only with "Assigned Vehicle" ticked.

[assigned_driver_assigned_vehicle.txt](https://github.com/cuba-platform/cuba/files/2509816/assigned_driver_assigned_vehicle.txt)
[assigned_vehicle.txt](https://github.com/cuba-platform/cuba/files/2509817/assigned_vehicle.txt)

Data model (relevant fields):

taxi$DriverDeviceInfo
Table: TAXI_DRIVER_DEVICE_INFO

Property | Column | Type | Description | Cardinality | Annotations
-- | -- | -- | -- | -- | --
device | DEVICE_ID | taxi$Device | Device | 1:1 | cuba.persistent

taxi$Device
Table: TAXI_DEVICE

Property | Column | Type | Description | Cardinality | Annotations
-- | -- | -- | -- | -- | --
id | ID | java.util.UUID | ID |   | cuba.persistent; cuba.primaryKey; cuba.system
driver |   | taxi$Driver | Current Driver | 1:1 | cuba.persistent
assignedVehicle |   | taxi$Vehicle | Assigned Vehicle | 1:1 | cuba.persistent

taxi$Driver
Table: TAXI_DRIVER

Property | Column | Type | Description | Cardinality | Annotations
-- | -- | -- | -- | -- | --
id | ID | java.util.UUID | ID |   | cuba.persistent; cuba.primaryKey; cuba.system
assignedDevice | ASSIGNED_DEVICE_ID | taxi$Device | Assigned Device | 1:1 | cuba.persistent

taxi$Vehicle
Table: TAXI_VEHICLE

Property | Column | Type | Description | Cardinality | Annotations
-- | -- | -- | -- | -- | --
id | ID | java.util.UUID | ID |   | cuba.persistent;  cuba.primaryKey;  cuba.system
device | DEVICE_ID | taxi$Device | Device | N:1 | cuba.persistent

The data model of relevant entities attached.
[datamodel_part.docx](https://github.com/cuba-platform/cuba/files/2509912/datamodel_part.docx)

**Comment by alexbudarov** [comment]
Infamous "you must use left join for variables used in OR condition" EclipseLink problem as I understand this.

**Comment by jreznot** [comment]
Yes, all the relations with . (like user.group) treated as inner join in EclipseLink.
